components/audit_log_filter/audit_log_filter.cc

```diff
@@ -751,9 +745,14 @@ std::string AuditLogFilter::get_sql_text(MYSQL_THD thd) {
 
   // According to mysql_thd_attributes_imp::get, "sql_text" is a String*.
   String *s = reinterpret_cast<String *>(hstr);
```

Comment:
After taking another look. I think we should NOT be using hack with explicitly converting "string handle/my_h_string" returned by mysql_thd_attributes to String pointer. Instead, IMO, we should be using appropriate service that works with such string handles.
To get access to string contents we can use `mysql_string_get_data_in_charset::get_data()` for example.
And to dispose string handle/underlying String object we should use `mysq_string_factory::destroy()`.